### PR TITLE
feat: standardize make extract_translations

### PR DESCRIPTION
**Pull request by shadinaif**
feat: standardize make extract_translations

Refs: FC-0012 OEP-58

This PR prepares the repository to comply with [openedx-translations](https://github.com/openedx/openedx-translations) by doing the following:

* Move translation directory from `openassessment/locale` to `openassessment/conf/locale`
* Create a symbolic link from `openassessment/locale` to `openassessment/conf/locale`
* Fix the translation command `make extract_translations` to use `django-admin` from inside `openassessment` directory instead of adding too many `--ignore`

- [x] Tested on local devstack palm to ensure that everything renders fine
- [x] Tested on local devstack palm to ensure that local translation still works

References
----------

This pull request is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

Check the links above for full information about the overall project.

Internalization is being rearchitected in Open edX Python, XBlock, Micro-frontend, and other projects. There are a number of immediately visible changes:
 - Remove source and language translations from the repositories, hence no `.json`, `.po` or `.mo` files will be committed into the repos.
 - Add standardized `make extract_translations` in all repositories
 - Push user-facing messages strings into [openedx/openedx-translations](https://github.com/openedx/openedx-translations/).
 - Integrate root repositories with [openedx/openedx-atlas](https://github.com/openedx/openedx-atlas/) to pull translations on build/deploy time

Breaking Changes
----------------

One of the primary goals of the project is to **avoid breaking changes**. If you noticed any suspicious code, please raise your concern. But before that, please know the strategy we're following to avoid breaking changes

**For XBlocks:**
 - The standard translation path must be `conf/locale`. Therefore, we are creating a link from `conf/locale` to `translations` so Atlas can find the path without disrupting the current way of having translations locally within the XBlocks
 - [openedx-translations](https://github.com/openedx/openedx-translations) will have a related PR that adds the XBlock to the pipeline. This will not affect the current way of managing/updating translations
 - At the end of the project, a clear change log will be added and all translations will be handled by Atlas. Thus, the local translation will be removed from the repo within the version bump
 - A notification for the community will be published to ensure that everyone knows why translations directories are removed from all repos

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @shadinaif!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by shadinaif** [comment]
@OmarIthawi @brian-smith-tcril this is ready for your review please

**Comment by shadinaif** [comment]
@mattcarter please take a look at this one

File: settings/base.py [hunk 1]
```diff
@@ -143,7 +143,7 @@
 
 BASE_DIR = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
 
-LOCALE_PATHS = [os.path.join(BASE_DIR, "openassessment", "locale")]
+LOCALE_PATHS = [os.path.join(BASE_DIR, "openassessment", "conf", "locale")]
```
Comment (OmarIthawi) [inline settings/base.py:146]: If I'm not mistaken, this is the correct configuration:

```suggestion
LOCALE_PATHS = [os.path.join(BASE_DIR, "openassessment", "conf/locale")]
```

Comment (shadinaif) [reply settings/base.py:146]: The result is the same for both @OmarIthawi . I guess the error will disappear if we run the test again, it's codecov issue

**Review by brian-smith-tcril — APPROVED** [review]

File: settings/base.py (hunk 1 above)
Comment (OmarIthawi) [reply settings/base.py:146]: Oh, my bad. It's a `join` operation. Looks great.

**Review by OmarIthawi — APPROVED** [review]

**Comment by brian-smith-tcril** [comment]
@shadinaif looks like non-codecov related tests are failing now, I tried re-running them a few times and there are still errors. Any ideas?

**Comment by itsjeyd** [comment]
🗒️ Blocks https://github.com/openedx/openedx-translations/pull/244

**Comment by shadinaif** [comment]
@brian-smith-tcril I've fixed it. Now it's passing locally, it should pass here too

**Comment by openedx-webhooks** [comment]
@shadinaif 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/1932&entry.1484655318=https://github.com/shadinaif&entry.752974735=2023-04-27+07:04&entry.1917517419=2023-05-09+14:39&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**